pkg/transformers/image_test.go

```diff
@@ -251,7 +253,181 @@ func TestImageTransformer(t *testing.T) {
 		{Name: "gcr.io:8080/my-project/my-cool-app", NewName: "my-cool-app"},
 		{Name: "postgres", NewName: "my-postgres", NewTag: "v3"},
 		{Name: "docker", NewName: "my-docker", Digest: "sha256:25a0d4b4a4c0eb97a1aabb8e29f18e917d05abfe1b7a7c07857230879ce7d3d3"},
-	})
+	}, defaultTransformerConfig.Images)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	err = it.Transform(m)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if !reflect.DeepEqual(m, expected) {
+		err = expected.ErrorIfNotEqual(m)
+		t.Fatalf("actual doesn't match expected: %v. Actual %+v", err, m)
+	}
+}
+
+func TestImageTransformerCustomConfig(t *testing.T) {
```

Comment:
This is nice, and adequate.

If you'd like, you could add a test in the form of
https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/target/customconfig_test.go#L87

In these tests one specifies the input resources,
the config, and the output resources in easier to read YAML form
without all this `map[string]interface{}` noise. 

Theses are useful as examples, because they're more readable.

Reply:
These codes exist before the patch. See https://github.com/kubernetes-sigs/kustomize/pull/841/files#diff-ea2cf8f5d028d0629e9b68593ef04be6L255

We may improve the readability in a separate PR and focus on function enhancement in this one. What do you think?